matching-web/src/main/java/com/graphhopper/matching/http/MatchServlet.java

```diff
@@ -114,8 +117,14 @@ public void doPost(HttpServletRequest httpReq, HttpServletResponse httpRes)
         if (!matchGHRsp.hasErrors()) {
             try {
                 FlagEncoder encoder = hopper.getEncodingManager().getEncoder(vehicle);
-                MapMatching matching = new MapMatching(hopper.getGraphHopperStorage(), locationIndexMatch, encoder);
-                matching.setMaxVisitedNodes(maxVisitedNodes);
+                AlgorithmOptions opts = AlgorithmOptions.start()
+                        .algorithm(Parameters.Algorithms.DIJKSTRA_BI)
+                        .traversalMode(hopper.getTraversalMode()).flagEncoder(encoder)
+                        .weighting(new FastestWeighting(encoder))
+                        .maxVisitedNodes(maxVisitedNodes).hints(new HintsMap()
+                                .put("weighting", "fastest").put("vehicle", encoder.toString()))
+                        .build();
```

Comment:
The best would be to remove `.weighting(...)` and just do .put("weighting", weightingFromURL)